### DISCUSSION TITLE
Elasticsearch: Query variable with multiple fields

### DISCUSSION THREAD

**Discussion by 4quarks** [grafana/grafana#58406] [Data Sources]
Dear Grafana team,
I would like to know if it is possible to query a variable using multiple fields. 
i.e. {"find": "terms", "field": "data.name", "field": "data.name_short", "size": 1000}
I would like to use the information from two fields in the same variable. 
Thanks!

**Comment by torkelo** [comment]
How would you use this variable?

**Comment by 4quarks** [comment]
It would be used in the same way as the normal variable. The only thing is that it would collect information from two different fields. Another possibility would be to combine two variables or have multiple queries:
i.e. 
{"find": "terms", "field": "data.name"} + {"find": "terms", "field": "data.name_short"}

**Comment by Elfo404** [comment]
Combining 2 variables or having 2 queries for the same variable is not currently possible.

We need to check if this is achievable with a single Terms query (https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-terms-query.html#query-dsl-terms-lookup) and probably we'll need to update how the query is sent/parsed in the elasticsearch plugin

**Comment by IamShubhamGupto** [comment]
Hey @4quarks 

Does the query {"find": "terms", "field": "data.name"} return any values? Im also running a terms query but I get None all the time

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
